### DISCUSSION TITLE
DOM은 HTML에서 그려야할까요, JS에서 그려야 할까요?

### DISCUSSION THREAD

**Discussion by byhhh2** [woowacourse-study/2022-code-review-study#25] [Lotto] [answered]
[PR 120](https://github.com/woowacourse/javascript-lotto/pull/120#discussion_r817445951)을 참고 

- 작성자의 입장 : 유저의 액션에 의해 DOM을 추가하거나 변경하는 것은 JS가 담당, 유저의 액션과 관계없이 항상 존재해야하는 DOM 요소는 HTML에서 작성해야 한다고 생각
- 리뷰어의 입장 : JS자체가 DOM을 조작하기 위해 사용하는 언어인데 JS에서 DOM조작을 최소화하기보다는 좀 더 효과적으로 JS를 활용해 DOM을 조작하는게 좋은 것 같다고 생각 (`template.js`를 사용하자)

이 토론을 위해서는 *서비스가 거대하지 않다*는 가정이 있어야 할 것 같습니다.

**Comment by wonsss** [comment]
저도 유저의 액션과 관계없이 항상 존재해야 하는 DOM 요소는 HTML에서 작성해도 괜찮을 것 같아요.
근데 그 양은 많지 않을 것 같아요.

만약 서비스가 거대하다는 가정을 하게 된다면, DOM조작을 JS를 통해 많이 하게 되겠죠.
그리고 JS로 DOM 조작하는 것도 꽤 복잡할 것 같다는 생각이 듭니다.

주제와 약간 벗어난 이야기이긴 하지만,
이와 관련해서 찾다보니, Web components라는 기술에 대해 알게 됐는데,
재사용하는 엘리먼트들과 메서드를 묶어서 커스텀 태그로 축약해서 쓸 수 있는 기술인 것 같아요.
한 번 써보고 싶네요.
https://developer.mozilla.org/ko/docs/Web/Web_Components

**Comment by moonheekim0118** [comment, marked as answer]
저는 JS 에서 그려줘야 한다고 생각해요.
작성자는 '유저의 액션과 관계없이 항상 존재해야하는 DOM 요소는 HTML에서 작성해야 한다'라고 했는데, 해당 DOM요소가 언제까지 유저의 액션과 관계 없을지 우리가 예측 할 수 없지 않을까요 ?! 하루아침에 명세가 바뀔 수도 있고요.. 👀..일단 일부분이라도 JS로 작성하기로 했다면 전부 JS로 작성하는게 맞다고 생각해요.
